### PR TITLE
feat/billing-plan

### PR DESCRIPTION
**Pull request by Ayobamidele**
## Description

This pull request introduces a new endpoint `GET /api/v1/organizations/{id}/billing-plans` that allows users to retrieve all billing plans associated with a specified organization. The endpoint responds with a JSON object containing the status of the request, a message, and an array of billing plans, each with its ID, name, price, and features.

### Example Response:

```json
{
  "status": 200,
  "message": "plans fetched successfully",
  "data": {
    "plans": [
      {
        "id": "basic",
        "name": "Basic Plan",
        "price": 10.00,
        "features": ["Feature 1", "Feature 2"]
      },
      {
        "id": "premium",
        "name": "Premium Plan",
        "price": 20.00,
        "features": ["Feature 1", "Feature 2", "Feature 3"]
      }
    ]
  }
}
```
Related Issue (Link to issue ticket)
[#321](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/321)

**Motivation and Context**

This change is required to provide organizations with the ability to view available billing plans. It addresses the need for enhanced visibility of billing options, allowing organizations to make informed decisions about subscription levels and features they may want to use.

**How Has This Been Tested?**

The new endpoint has been tested using Postman to ensure that the correct billing plans are fetched based on valid organization IDs. Testing included the following:

    

- [x] Sending requests with valid authorization tokens.
- [x] Verifying the response format, status code, and data integrity.
- [x] Ensuring that appropriate status codes are returned for unauthorized tokens

Screenshot 2024-07-24 125004
Screenshot 2024-07-24 125025

Types of changes

   - [ ] Bug fix (non-breaking change which fixes an issue)
   - [x] New feature (non-breaking change which adds functionality)
   - [ ] Breaking change (fix or feature that would cause existing functionality to change)

Checklist:

  -  [x] My code follows the code style of this project.
  -  [ ] My change requires a change to the documentation.
  -  [ ] I have updated the documentation accordingly.
  -  [x] I have read the CONTRIBUTING document.
  -  [x] I have added tests to cover my changes.
  -  [x] All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Pull recent changes from tye dev branch and look through the codebase properly most of those functions you are creating in deps.py are already being taken care of in api/v1/servuces/user.py

Please read through the code and follow proper structure

Also create a service for whatever feature you are working on because mentors won't approve of this

**Comment by zxenon-sigmund** [comment]
> Pull recent changes from tye dev branch and look through the codebase properly most of those functions you are creating in deps.py are already being taken care of in api/v1/servuces/user.py
> 
> Please read through the code and follow proper structure
> 
> Also create a service for whatever feature you are working on because mentors won't approve of this

@Ayobamidele

**Comment by zxenonx** [comment]
> Pull recent changes from tye dev branch and look through the codebase properly most of those functions you are creating in deps.py are already being taken care of in api/v1/servuces/user.py
> 
> Please read through the code and follow proper structure
> 
> Also create a service for whatever feature you are working on because mentors won't approve of this

@Ayobamidele

**Comment by Ayobamidele** [comment]
@joboy-dev @zxenonx  I've made the fixes

File: api/v1/routes/auth.py [hunk 1]
```diff
@@ -130,9 +130,9 @@ def refresh_access_token(request: Request, response: Response, db: Session = Dep
 
     return response
     
-
 # Protected route example: test route
 @auth.get("/admin")
 def read_admin_data(current_admin: Annotated[User, Depends(get_current_admin)]):
     return {"message": "Hello, admin!"}
```
Comment (zxenonx) [inline api/v1/routes/auth.py:137]: What is this endpoint for?

File: api/v1/routes/bill_plan.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from sqlalchemy.orm import Session
+from api.v1.models.user import User
+from api.v1.services.plan import billing_plan_service
+from api.db.database import get_db
+from api.utils.json_response import JsonResponseDict
+from api.v1.services.user import user_service
+
+bill_plan = APIRouter(prefix='/organizations', tags=['Billing-Plan'])
+
+@bill_plan.get('/12345/billing-plans')
```
Comment (zxenonx) [inline api/v1/routes/bill_plan.py:15]: Why this route? '/12345/billing-plans'

File: tests/v1/test_plan.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        updated_at=datetime.now(timezone.utc)
+    )
+    mock_db_session.query.return_value.filter.return_value.first.return_value = mock_user
+
+    # mock_db_session.return_value.__enter__.return_value = mock_user
+    # mock_user_service.hash_password.return_value = "hashed_password"
+    # mock_db_session.add.return_value = None
+    # mock_db_session.commit.return_value = None
+    # mock_db_session.refresh.return_value = None
+
```
Comment (zxenonx) [inline tests/v1/test_plan.py:66]: we can remove this

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @Ayobamidele 
I left a few comments.

**Comment by joboy-dev** [comment]
@Ayobamidele please attend to requested changes

File: api/v1/routes/auth.py (hunk 1 above)
Comment (Ayobamidele) [reply api/v1/routes/auth.py]: Updated code with fix

File: tests/v1/test_plan.py (hunk 3 above)
Comment (Ayobamidele) [reply tests/v1/test_plan.py]: Thanks

File: api/v1/routes/bill_plan.py (hunk 2 above)
Comment (Ayobamidele) [reply api/v1/routes/bill_plan.py]: I've cleared them out

**Comment by Ayobamidele** [comment]
@zxenonx @joboy-dev I've made the changes.

**Comment by Ayobamidele** [comment]
@buka4rill I've corrected the PR name

**Pull request closed without merging**